src/token.sw

```diff
@@ -25,3 +27,66 @@ pub fn force_transfer(amount: u64, asset_id: ContractId, contract_id: ContractId
         tr r3 r1 r2;
     }
 }
+
+/// Transfer `amount` coins of type `asset_id` to address `recipient`.
+pub fn transfer_to_output(amount: u64, asset_id: ContractId, recipient: Address) {
+    // note: if tx format changes, the magic number "48" must be changed !
+    // TransactionScript outputsCount has a 48 byte(6 words * 8 bytes) offset
+    // Transaction Script: https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#transactionscript
+    // Output types: https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#output
+    const OUTPUT_LENGTH_LOCATION = 48;
+    const OUTPUT_VARIABLE_TYPE = 4;
+
+    // get length of outputs from TransactionScript outputsCount:
+    let length: u8 = asm(outputs_length, outputs_length_ptr: OUTPUT_LENGTH_LOCATION) {
+        lw outputs_length outputs_length_ptr i0;
+        outputs_length: u8
```

Comment:
Since the `outputsCount` value is a byte, you want to use `lb` to load it, rather than `lw` which will load 7 extra bytes from the header too.

Reply:
You could do either since things are word-aligned.

Reply:
OK, I missed that detail.  Why are they documented as smaller types when they're all aligned?  They might as well be all `uint64`s.

Reply:
`lb` is _more_ correct though, I think, since it isn't actually part of the spec (or at least a part we'd want to rely on) that the rest of the bytes will be zeroed out and/or ignored.

Reply:
I replaced `lw` with `lb`.